hypothesis: 1.14.0 -> 3.1.0

DamienCassou opened the pull request:
###### Things done:
- [x] Tested using sandboxing (`nix-build --option build-use-chroot true` or [nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS)
- [x] Built on platform(s): NixOS
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

nox-review is still not finished, I fixed everything that got reported until now.

Question: should I continue to fix every problem found by nox-review in the process of updating hypothesis, or should I better create an hypothesis3 derivation? I'm ok with both.

/cc @domenkozar @vcunat @edolstra

lancelotsix commented:
I think the best option would be to fix errors as much as possible (which could require some work). An alternative would be to 1) rename `hypothesis` into `hypothesis1` 2) make sure that all derivations that did depend on  `hypothesis`  now depend on `hypothesis1` and then 3) introduce `hypothesis` with the desired version.

`hypothesis` could be an alias to `hypothesis3` but I am not sure if this will be the best option on the long run (we tend to prefer one version maintained).

Considering the amount of changes, the `python-wip` (build by this hydra instance https://headcounter.org/hydra/jobset/iElectric/python-wip) could be a good first destination.

Also, could it be possible to have 1 commit for each package update (makes it easier to track down)?

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -3890,17 +3891,19 @@ in modules // {
 
     propagatedBuildInputs = with self; [ tox py eventlet ];
 
+    disabled = isPy3k;
```
why disabled?

DamienCassou committed:
glance: 11.0.0 -> 11.0.1

DamienCassou committed:
betamax: fix test dependencies in Python 3.5

DamienCassou committed:
gmusicapi: fix dependency to gpsoauth

There is a newer version of gmusicapi depending on a newer version of
gpsoauth but I had problems building it. I leave updating gmusicapi as
an exercise to the maintainer.

DamienCassou committed:
gpsoauth: 0.0.4 -> 0.2.0

DamienCassou committed:
hypothesis: 1.14.0 -> 3.1.0

I kept version 1.14.0 in hypothesis1 because at least one project
depends on it.

DamienCassou committed:
kafka: init at 0.9.5

DamienCassou committed:
oslo-utils: 2.6.0 -> 2.8.0

DamienCassou committed:
oslo-middleware: 2.9.0 -> 3.0.0

DamienCassou committed:
oslo-messaging: 2.7.0 -> 3.0.0

DamienCassou committed:
oslo-service: 0.10.0 -> 0.13.0

DamienCassou committed:
oslo-config: 2.5.0 -> 2.7.0

DamienCassou committed:
pycryptodome: init at 3.4

DamienCassou committed:
detox: 0.9.3 -> 0.10.0

DamienCassou commented:
> I think the best option would be to fix errors as much as possible

done. `nox-review wip` says everything is fine.

> Considering the amount of changes, the `python-wip` could be a good first destination.

I'm done with the changes. Do you still think it is important to go through `python-wip`?

> Also, could it be possible to have 1 commit for each package update?

sure, that was planned. Done.

lancelotsix commented:
I’m running nox-review on my side, this should be enough (no need for the entire hydra job).

FRidh commented:
@domenkozar this PR also updates oslo/openstack. ok?

domenkozar commented:
Yeah we need to test that openstack still works (I have a PR that has a NixOS test)

lancelotsix commented on pkgs/top-level/python-packages.nix:
```diff
@@ -16164,6 +16211,19 @@ in modules // {
+    meta = {
+      homepage = http://www.pycryptodome.org;
+      description = "Self-contained Python package of low-level cryptographic primitives";
+      license = licences.bsd2;
```
s/licences/licenses/

FRidh commented:
@domenkozar do you think that test can be merged directly so we can test whether we can merge this?
If not, I suggest @DamienCassou splits this PR into two, and follows @lancelotsix advice of creating a `hypothesis_1` and `hypothesis_3`.

DamienCassou commented:
I managed to run `nox-review wip` some days ago, but `nox-review pr` gives me errors I can't reproduce with `nix-build`.

DamienCassou commented:
Ok, I merged #13876. This PR is useless now.
